test/unit/sentinel/PCVSentinel.test.ts

```diff
@@ -0,0 +1,216 @@
+    const balanceGuardFactory = await ethers.getContractFactory('BalanceGuard');
+    const multiActionGuardFactory = await ethers.getContractFactory('MultiActionGuard');
+    const reEntrancyGuardFactory = await ethers.getContractFactory('ReEntrancyGuard');
+
+    pcvSentinel = await (await pcvSentinelFactory.deploy(core.address)).deployed();
+    noOpGuard = await (await noOpGuardFactory.deploy()).deployed();
+    balanceGuard = await (await balanceGuardFactory.deploy()).deployed();
+    multiActionGuard = await (await multiActionGuardFactory.deploy()).deployed();
+    reEntrancyGuard = await (await reEntrancyGuardFactory.deploy()).deployed();
+
+    // To deploy a contract, import and use the contract factory specific to that contract
+    // note that the signer supplied is optional
+  });
+
+  // Try and do as much deployment in beforeEach, and as much testing in the actual functions
+  describe('sentinel setup and view funcs', async () => {
+    it('has no guards upon deployment', async () => {
+      expect((await pcvSentinel.allGuards()).length).to.equal(0);
+    });
+
+    it('reports guards after they are added', async () => {
+      await pcvSentinel.connect(impersonatedSigners[guardianAddress]).knight(noOpGuard.address);
+      await pcvSentinel.connect(impersonatedSigners[guardianAddress]).knight(balanceGuard.address);
+      await pcvSentinel.connect(impersonatedSigners[guardianAddress]).knight(multiActionGuard.address);
+      await pcvSentinel.connect(impersonatedSigners[guardianAddress]).knight(reEntrancyGuard.address);
+
+      expect((await pcvSentinel.allGuards()).length).to.equal(4);
+      expect(await pcvSentinel.isGuard(noOpGuard.address)).to.equal(true);
+      expect(await pcvSentinel.isGuard(balanceGuard.address)).to.equal(true);
+      expect(await pcvSentinel.isGuard(multiActionGuard.address)).to.equal(true);
+      expect(await pcvSentinel.isGuard(reEntrancyGuard.address)).to.equal(true);
+      expect(await pcvSentinel.isGuard(userAddress)).to.equal(false);
+    });
+  });
+
+  describe('sentinel with no-op guard', async () => {
+    it('adds and checks and protecs', async () => {
+      await pcvSentinel.connect(impersonatedSigners[guardianAddress]).knight(noOpGuard.address);
+      expect(await pcvSentinel.isGuard(noOpGuard.address)).to.equal(true);
+      await expect(pcvSentinel.protec(noOpGuard.address)).to.emit(pcvSentinel, 'Protected');
+    });
+  });
+
+  describe('sentinel with basic condition check', async () => {
+    it('adds and checks and protecs', async () => {
+      await pcvSentinel.connect(impersonatedSigners[guardianAddress]).knight(balanceGuard.address);
+      expect(await pcvSentinel.isGuard(balanceGuard.address)).to.equal(true);
+      await expect(pcvSentinel.protec(balanceGuard.address)).to.emit(pcvSentinel, 'Protected');
+    });
+
+    it('reverts if condition not met', async () => {
+      await pcvSentinel.connect(impersonatedSigners[guardianAddress]).knight(balanceGuard.address);
```

Comment:
add this into a beforeEach hook instead of duplicating this logic across multiple tests

Reply:
Which ones we want knighted aren't the same across tests, otherwise I would